packages/@divvi/mobile/src/viem/saga.ts

```diff
@@ -121,6 +102,15 @@ export function* sendPreparedTransactions(
       hash
     )
 
+    if (i === 0) {
+      yield* call(submitDivviReferralIfNeeded, {
+        walletAddress: wallet.account.address,
+        txHash: hash,
+        chainId,
+        transactionRequest: preparedTransaction,
+      })
```

Comment:
This should be non-blocking here I think, since it's more of a background task.
And errors happening in it (though I see they are wrapped in try/catch there) shouldn't disturb the existing saga.

Reply:
its no longer a time consuming function